build/operator/Dockerfile

```diff
@@ -1,20 +1,54 @@
-FROM quay.io/centos/centos:stream9
+# Define build-time arguments
+ARG BUILD_ARCH=amd64            # Default value for architecture is amd64
+ARG GO_VERSION="1.23.3"         # Default Go version is 1.23.3
```

Comment:
might worth to auto take it from go.mod
there are few network repos that doing something like that

Reply:
Good suggestion! I'll look into implementing that.

Reply:
I made the changes according to the script [install-go.sh](https://github.com/kubevirt/cluster-network-addons-operator/blob/main/hack/install-go.sh). With the current setup, we are compiling and building the manager binary using the Go version specified above.

Please note that the script [go-version.sh](https://github.com/kubevirt/cluster-network-addons-operator/blob/main/hack/go-version.sh) returns the Go version in major.minor format. Additionally, the command curl -L -s "https://go.dev/dl/?mode=json" does not list Go 1.21, as it's archived. Before Go 1.22, we would need to download it directly from the archive. However, the install-go.sh script currently downloads the latest Go version and proceeds to compile with it.

I’ve made the same update in the Dockerfile to ensure consistency.